Modules/Video/BridgeOpenCV/include/itkOpenCVImageBridge.hxx

```diff
@@ -251,14 +297,135 @@ template <typename TInputImageType>
 cv::Mat
 OpenCVImageBridge::ITKImageToCVMat(const TInputImageType * in, bool force3Channels)
 {
-  // Extra copy, but necessary to prevent memory leaks
-  IplImage * temp = ITKImageToIplImage<TInputImageType>(in, force3Channels);
-  cv::Mat    out = cv::cvarrToMat(temp, true);
-  cvReleaseImage(&temp);
+  using namespace cv;
+
+  // Typedefs
+  using ImageType = TInputImageType;
+  using InputPixelType = typename ImageType::PixelType;
+  using ValueType = typename itk::NumericTraits<InputPixelType>::ValueType;
+
+  //
+  // Make sure input isn't null, is 2D or 1D, and is scalar or RGB
+  //
+  if (!in)
+  {
+    itkGenericExceptionMacro("Input is nullptr");
+  }
+
+  typename ImageType::RegionType region = in->GetLargestPossibleRegion();
+  typename ImageType::SizeType   size = region.GetSize();
+
+  if (ImageType::ImageDimension > 2)
+  {
+    bool IsA2DImage = false;
+    for (unsigned int dim = 2; (dim < ImageType::ImageDimension) && !IsA2DImage; dim++)
+    {
+      if (size[dim] != 1)
+      {
+        IsA2DImage = true;
+      }
+    }
+    if (IsA2DImage)
+    {
+      itkGenericExceptionMacro("OpenCV only supports 2D and 1D images");
+    }
+  }
+
+  unsigned int inChannels = itk::NumericTraits<InputPixelType>::MeasurementVectorType::Dimension;
+  if (inChannels != 1 && inChannels != 3)
+  {
+    itkGenericExceptionMacro("OpenCV only supports scalar and 3-channel data");
+  }
+
+  unsigned int outChannels = inChannels;
+  //
+  // Set up the output image
+  //
+  Mat          tmp;
+  unsigned int w = static_cast<unsigned int>(size[0]);
+  unsigned int h = static_cast<unsigned int>(size[1]);
+
+  //
+  // set the depth correctly based on input pixel type
+  //
```

Comment:
These empty comment lines before  and after the actual comment are not exactly needed. They reduce code density necessitating more scrolling. Please remove them.

Reply:
I actually was not the one to format the comments like that. It was in the original code so I thought I should leave it like that. I do agree however that this is not a great coding style. I also realised that I left one include (<iostream>) that was only for debug when I was testing my changes.

I am willing to  make these corrections. However, this is the first time I submit a pull request and I am not sure how to submit the changes.

If I understand correctly, you also want me to ammend my commit message to include the comment about ideas for further improvements. Is that correct?

Reply:
Corrections: amend the commit and then force-push to the same branch (`opencv-bridge-update`).

Ideas for future improvement were my message to you, and should not be included in commit message. Thanks for working on this!

Reply:
Hi @dzenanz, the cleanup and correction of comment format is done. I also removed the useless comments saying "return result".